controller/src/lib.rs

```diff
@@ -20,62 +24,120 @@ pub mod plan;
 pub mod run;
 mod scheduler;
 
-pub async fn run_scheduler(nats: TypedNats, state: StateHandle) -> NeverResult {
-    let scheduler = Scheduler::new(state);
-    let mut schedule_request_sub = nats.subscribe(ScheduleRequest::subscribe_subject()).await?;
-    tracing::info!("Subscribed to spawn requests.");
+async fn schedule_spawn(
+    backend: &BackendResource,
+    scheduler: &Scheduler,
+    cluster: &ClusterName,
+    nats: &TypedNats,
+) -> anyhow::Result<ScheduleResponse> {
+    tracing::info!(spawn_request=?backend, "Got spawn request");
+    Ok(match scheduler.schedule(cluster, Utc::now()) {
+        Ok(drone_id) => {
+            let timer = Timer::new();
+            let spawn_request = backend.schedule(cluster, &drone_id);
+            match nats.request(&spawn_request).await {
+                Ok(true) => {
+                    tracing::info!(
+                        duration=?timer.duration(),
+                        backend_id=%spawn_request.backend_id,
+                        %drone_id,
+                        "Drone accepted backend."
+                    );
 
-    while let Some(schedule_request) = schedule_request_sub.next().await {
-        tracing::info!(spawn_request=?schedule_request.value, "Got spawn request");
-        let result = match scheduler.schedule(&schedule_request.value.cluster, Utc::now()) {
-            Ok(drone_id) => {
-                let timer = Timer::new();
-                let spawn_request = schedule_request.value.schedule(&drone_id);
-                match nats.request(&spawn_request).await {
-                    Ok(true) => {
-                        tracing::info!(
-                            duration=?timer.duration(),
-                            backend_id=%spawn_request.backend_id,
-                            %drone_id,
-                            "Drone accepted backend."
-                        );
+                    nats.publish(&WorldStateMessage {
```

Comment:
it'd be nice to enforce that ResourceRequests being accepted corresponded with a StateUpdate in the typesystem somehow.